### PR TITLE
Evac merchant can food quest (attempt 3)

### PR DESCRIPTION
**Pull request by DavisBroda**
#### Summary

SUMMARY: Content "Adds new evac merchant mission 'canned food'"

#### Purpose of change

Evac merchant questline is a bit short, anf could use some additional content. Additionally the evac center storage room always felt a little empty, so adding this mission allows it to be filled up a bit. 

#### Describe the solution

Mission requires the player to get 40 items of canned food for the merchant. Once completed 40 items of canned food are spawned in the evac center storage room. These spawned items are fixed, and completely uneffected by which specific types of canned food the player uses to complete the quest.

#### Describe alternatives you've considered

Considered trying to make the items spawned in the storage room match those turned in by the player, but did not know the project well enough to implement that. 

#### Additional context

Note: there was a prior pull request for this change, but the freeze for 0.D and moving it's base branch from master to dev caused a really difficult merge. Easier to just move the relevant changes to a new branch, close the old request, and open a new one. Second attempt was closed due to confusion as to which branch to target.

### CODE REVIEW

File: src/mission.cpp [hunk 1]
```diff
@@ -249,6 +252,45 @@ void mission::wrap_up()
...
+            }
+
+            u.consume_items( comps );
+
+            if( remove_container ) {
+                std::vector<item_comp> container_comp = std::vector<item_comp>();
+                if( empty_container != "null" ) {
+                    container_comp.push_back( item_comp( empty_container, type->item_count ) );
+                    u.consume_items( container_comp );
+
```
Comment (Night-Pryanik) [inline src/mission.cpp]: Empty lines here and below.

File: src/mission.cpp [hunk 2]
```diff
@@ -249,6 +252,45 @@ void mission::wrap_up()
     u.on_mission_finished( *this );
     std::vector<item_comp> comps;
     switch( type->goal ) {
+        case MGOAL_FIND_ITEM_GROUP: {
+            inventory tmp_inv = u.crafting_inventory();
+            std::vector<item *> items = std::vector<item *>();
+            tmp_inv.dump( items );
+            Group_tag grp_type = type->group_id;
+            itype_id container = type->container_id;
+            bool specific_container_required = container.compare( "null" ) != 0;
```
Comment (Night-Pryanik) [inline src/mission.cpp]: Have you tried `container != "null"`?

File: src/mission.cpp [hunk 3]
```diff
@@ -344,6 +411,50 @@ bool mission::is_complete( const int _npc_id ) const
...
 
+void mission::get_all_item_group_matches( std::vector<item *> &items,
+        Group_tag &grp_type, std::map<itype_id, int> &matches,
+        const itype_id &required_container, const itype_id &actual_container,
+        bool &specific_container_required )
+{
+    for( std::vector<int>::size_type i = 0; i < ( items ).size(); i++ ) {
+        item *itm = items[i];
+
+        bool correct_container = ( ( required_container ) == ( actual_container ) ) ||
```
Comment (Night-Pryanik) [inline src/mission.cpp]:
```suggestion
        bool correct_container = ( required_container == actual_container ) ||
```

File: src/mission.cpp [hunk 4]
```diff
@@ -344,6 +411,50 @@ bool mission::is_complete( const int _npc_id ) const
...
+            std::map<std::string, int>::iterator it = matches.find( itm->typeId() );
+            if( it != matches.end() ) {
+                it->second = ( it->second ) + 1;
+            } else {
+                matches.insert( std::make_pair( itm->typeId(), 1 ) );
+            }
+        }
+
+        //recursivly check item contents for target
+        if( ( itm->is_container() ) && !( itm->is_container_empty() ) ) {
```
Comment (Night-Pryanik) [inline src/mission.cpp]:
```suggestion
        if( itm->is_container() && !itm->is_container_empty() ) {
```

File: src/mission.cpp (hunk 1 above)
Comment (DavisBroda) [reply src/mission.cpp]: removed in latest commit

File: src/mission.cpp (hunk 2 above)
Comment (DavisBroda) [reply src/mission.cpp]: just tried it, it works. implemented in latest commit.

File: src/mission.cpp (hunk 3 above)
Comment (DavisBroda) [reply src/mission.cpp]: change implemented in latest commit

File: src/mission.cpp (hunk 4 above)
Comment (DavisBroda) [reply src/mission.cpp]: change implemented in latest commit

**Comment by DavisBroda** [comment]
CI test failure in 351e19c was the "clean_water_in_occupied_cooking_vessel" test. I don't think I touched any related to this.

**Pull request merged**